### ISSUE TITLE
Rethink grouping of alerts on dashboard

### ISSUE THREAD

**Issue by IzakMarais** [bosun-monitor/bosun#1130]
It seems the alert name is sometimes dropped form the group subject when alerts get grouped on the Bosun dashboard. I expected that the alert group subject heading should contain the alert name in addition to the tagset. However it looks like the alert name gets dropped, only the tagset is used.

According to @kylebrandt :

> "Need to think about it some more. The logic currently tries to make the biggest grouping it can with various constraints if I remember correctly. So sometimes it will group multiple different alerts if they share a group. Sometimes this is useful, for example multiple alerts about the same host=foo, other times it is not. For instance I have a group on my dashboard right now that just says critical - instance=1. So I'm considering that grouping by tagset only is outdated now that we support dependencies

> @izakmarais: Can you create an issue for this?

> Maybe we just shouldn't group at all on the dashboard anymore

I think that grouping alert instances of the same alert should be kept. 

We are still evaluating Bosun and do not yet have many rules running, so I cannot give accurate recommendations on advanced grouping strategies. I just didn't expect my alert names to be dropped.

**Comment by kylebrandt** [comment]
We ended up doing this. There is a threshold for grouping now so it is more of a (tons of alerts) thing.

**Comment by IzakMarais** [comment]
> We ended up doing this
Just to clarify the slightly cryptic "this" above, does the change include:
* new grouping threhold. Implies no grouping when total alerts on dashboard less then threshold?
* any change in the way groups names are displayed? The behaviour of when the alert name gets dropped from group the name?

**Comment by kylebrandt** [comment]
I don't think we changed the names, but we did add a new threshold where we don't group unless it less than the threshold. 

I don't think we did anything with naming IIRC. However, this is really meant as a backup safety mechanism. So if you are getting large groups frequently, it probably means your alerts should be scoped differently. So since they are rare, I don't think the getting the group naming just right really matters.
